persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java

```diff
@@ -23,60 +23,66 @@
  * Configured in: manytomany.cfg.xml
  */
 public class HibernateManyToManyAnnotationMainIntegrationTest {
-    private static SessionFactory sessionFactory;
+	private static SessionFactory sessionFactory;
 
-    private Session session;
+	private Session session;
 
-    @BeforeClass
-    public static void beforeTests() {
-        sessionFactory = HibernateUtil.getSessionFactory();
-    }
-
-    @Before
-    public void setUp() {
-        session = sessionFactory.openSession();
-        session.beginTransaction();
-    }
+	@BeforeClass
+	public static void beforeTests() {
+		sessionFactory = HibernateUtil.getSessionFactory();
+	}
 
-    @Test
-    public void givenData_whenInsert_thenCreatesMtoMrelationship() {
-        String[] employeeData = { "Peter Oven", "Allan Norman" };
-        String[] projectData = { "IT Project", "Networking Project" };
-        Set<Project> projects = new HashSet<Project>();
+	@Before
+	public void setUp() {
+		session = sessionFactory.openSession();
+		session.beginTransaction();
+	}
 
-        for (String proj : projectData) {
-            projects.add(new Project(proj));
-        }
-
-        for (String emp : employeeData) {
-            Employee employee = new Employee(emp.split(" ")[0], emp.split(" ")[1]);
-            assertEquals(0, employee.getProjects().size());
-            employee.setProjects(projects);
-            session.persist(employee);
-            assertNotNull(employee);
-        }
-    }
-
-    @Test
+	@Test
     public void givenSession_whenRead_thenReturnsMtoMdata() {
+		prepareData();
+       	@SuppressWarnings("unchecked")
+		List<Employee> employeeList = session.createQuery("FROM Employee").list();
         @SuppressWarnings("unchecked")
-        List<Employee> employeeList = session.createQuery("FROM Employee").list();
+		List<Project> projectList = session.createQuery("FROM Project").list();
         assertNotNull(employeeList);
+        assertNotNull(projectList);
+        assertEquals(2, employeeList.size());
+        assertEquals(2, projectList.size());
+        
         for(Employee employee : employeeList) {
             assertNotNull(employee.getProjects());
```

Comment:
Assert here that every employee has 2 projects assigned.

Reply:
This has been addressed